src/utils/http.js

```diff
@@ -30,27 +30,35 @@ export function stringifyGETParams(url, data) {
     return url;
 }
 
-export function handleStatus(response) {
-    if (response.status >= 200 && response.status < 300) {
-        return response;
-    }
-
-    var error = new Error(response.statusText);
-    error.response = response;
-
-    throw error;
-}
-
-
-export function handleBody(response) {
+export function handleResponse(response) {
     if (response.status === 202 || response.status === 204) {
         return Promise.resolve();
     }
 
     var contentType = response.headers.get('Content-Type') || '';
     var isJson = contentType.indexOf('application/json') > -1;
 
-    return isJson ? response.json() : Promise.resolve();
+    if (response.status >= 200 && response.status < 300) {
+        return isJson ? response.json() : Promise.resolve();
+    } else {
+        if (isJson) {
+            return response.json().then(function(json) {
+                const {error={}} = json;
+
+                var err = new Error(error.description || response.statusText);
```

Comment:
could you clean up the `var`s in here pls? I have no idea why there are `var`s in this project.